### PR TITLE
Fix bug in FiniteDateRange.is_disjoint

### PR DESCRIPTION
**Pull request by josh-lynch**
Previously if you called is_disjoint on a range containing either
date.min or date.max you would get an OverflowError.

### CODE REVIEW

**Review by greg-charvet-kraken — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## v4.7.1 - 2023-11-01
+
+- Adding explicit type for Optional OneToOneField [#113](https://github.com/octoenergy/xocto/pull/113)
```
Comment (pydanny) [inline CHANGELOG.md:5]: In semver, this is a 4.8.0 change, not 4.7.1

File: setup.py [hunk 2]
```diff
@@ -6,7 +6,7 @@
 
 REPO_ROOT = path.abspath(path.dirname(__file__))
 
-VERSION = "4.7.0"
+VERSION = "4.7.1"
```
Comment (pydanny) [inline setup.py:9]: Please change to 4.8.0

File: xocto/__init__.py [hunk 3]
```diff
@@ -1 +1 @@
-__version__ = "4.7.0"
+__version__ = "4.7.1"
```
Comment (pydanny) [inline xocto/__init__.py:1]: 4.8.0

File: xocto/ranges.py [hunk 4]
```diff
@@ -811,11 +811,17 @@ def is_disjoint(self, other: Range[datetime.date]) -> bool:
         other_start = other.start
         if other._is_left_inclusive:
             assert other_start is not None
-            other_start -= datetime.timedelta(days=1)
+            try:
+                other_start -= datetime.timedelta(days=1)
```
Comment (pydanny) [inline xocto/ranges.py:815]: Nothing needs to be changed here, merely a comment that  I rarely see `-=`. 🤷🏻

**Review by pydanny — CHANGES_REQUESTED** [review]
Looks really good! Just please update the version to `4.8.0`. Thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (josh-lynch) [reply CHANGELOG.md:5]: ✅

**Review by prestto — APPROVED** [review]

**Pull request merged**